pisa/core/map.py

```diff
@@ -975,6 +984,43 @@ def fluctuate(self, method, random_state=None, jumpahead=0):
                 error_vals[valid_mask] = np.sqrt(orig_hist[valid_mask])
                 error_vals[nan_at] = np.nan
             return {'hist': unp.uarray(hist_vals, error_vals)}
+        
+        if method == 'scaled_poisson':
+            random_state = get_random_state(random_state, jumpahead=jumpahead)
+            with np.errstate(invalid='ignore'):
+                orig_hist = self.nominal_values
+                sigma = self.std_devs
+
+                nan_at = np.isnan(orig_hist)
+                zero_at = orig_hist == 0.
+                valid_mask = ~nan_at
+                if np.any(sigma[valid_mask & ~zero_at] == 0.):
+                    logging.warn(
+                        "Some bins have non-zero counts but no assiciated error! "
+                        "All errors will be set to their Poisson expectation now. "
+                        "To avoid this warning, call `set_poisson_errors()` on the "
+                        "map or set non-zero errors manually."
+                    )
+                    sigma[valid_mask] = np.sqrt(orig_hist[valid_mask])
+                variance_valid = sigma[valid_mask]**2
+
+                if np.allclose(variance_valid, orig_hist[valid_mask], **ALLCLOSE_KW):
+                    scale_factor = 1.
+                else:
+                    scale_factor = variance_valid/orig_hist[valid_mask]
+                poisson_lambda = orig_hist[valid_mask]/scale_factor
+
+                hist_vals = np.empty_like(orig_hist, dtype=np.float64)
```

Comment:
not `dtype=FTYPE`?

Reply:
good point ... or even just leave off the dtype, as `empty_like` will pick up the dtype from `orig_hist`

Reply:
Then I'll change it in the original 'poisson' implementation as well, it does the same.

Reply:
Unless there was a reason to do it like that in the original?

Reply:
nope, I don't think so

Reply:
yep, probably an oversight